src/clib/pio_file.c

```diff
@@ -109,6 +109,7 @@ int PIOc_createfile(const int iosysid, int *ncidp, int *iotype,
         return PIO_ENOMEM;
 
     /* Fill in some file values. */
+    file->fh = -1;
```

Comment:
In this case fh is set on some procs and not on others. Initializing it to -1 ensures that when we debug the code we know that the handle is not set.